src/hashset.c

```diff
@@ -32,6 +32,36 @@
  * Functions and types are prefixed by "hashset", macros by "HASHSET". Internal
  * names don't use the prefix. Internal functions are 'static'.
  *
+ * Caution: directly storing non-pointer elements
+ * ----------------------------------------------
+ *
+ * Storing non-pointer values, such as pointer-sized integers (long), in a
+ * hashset is possible. Be careful when casting to and from `void *` though.
```

Comment:
Like we do with https://github.com/valkey-io/valkey/blob/unstable/src/dict.c#L76.

Reply:
In practice, we only store pointers though. In dict, we store integer values in the expire dict and double values in sorted sets(?) because we have a key as well in the dict entry, but in hashset we need the key and value together so there is no use case for storing only a set of integers or a set of floats, apart from in the unit tests.

We don't want a union that includes double and int64_t since it would make it 64 bits even on 32-bit platforms. We should only allow pointer-sized types in that case, such as `intptr_t i` and maybe `float` or `double` depending on platform. On top of that, using a union would make the usage a bit more complex.

Since the only use case is to unit tests, I'd rather skip using non-pointers in the unit tests instead.

Reply:
Then let's remove the large comment that rain added.

Reply:
Yeah, @SoftlyRaining WDYT?

We'd need to do some extra work to use some allocated objects in the unit tests then, or we can cheat and use longs and add comments about aliasing only in the unit test suite.

Reply:
I actually like this best.